### PR TITLE
[core] Async instantiate the CandyShop to get the required decimals & symbol

### PR DESCRIPTION
**Pull request by vanminh1701**

### CODE REVIEW

File: core/sdk/src/shop/sol/CandyShop.ts [hunk 1]
```diff
@@ -147,23 +184,16 @@ export class CandyShop extends BaseShop implements CandyShopAuctioneer, CandySho
...
-      currencyDecimals: settings?.currencyDecimals ?? DEFAULT_CURRENCY_DECIMALS,
-      priceDecimals: settings?.priceDecimals ?? DEFAULT_PRICE_DECIMALS,
-      priceDecimalsMin: settings?.priceDecimalsMin ?? DEFAULT_PRICE_DECIMALS_MIN,
-      volumeDecimals: settings?.volumeDecimals ?? DEFAULT_VOLUME_DECIMALS,
-      volumeDecimalsMin: settings?.volumeDecimalsMin ?? DEFAULT_VOLUME_DECIMALS_MIN,
-      mainnetConnectionUrl: settings?.mainnetConnectionUrl ?? DEFAULT_MAINNET_CONNECTION_URL,
-      connectionConfig: settings?.connectionConfig,
-      explorerLink: settings?.explorerLink ?? ExplorerLinkBase.SolanaFM
-    };
+  private constructor(candyShopAddress: web3.PublicKey, params: SolanaShopConstructorParams) {
```
Comment (haVincy) [inline core/sdk/src/shop/sol/CandyShop.ts:187]: I assume this PR will break backward compatibility? @RustySol @vanminh1701

File: core/sdk/src/shop/sol/CandyShop.ts [hunk 2]
```diff
@@ -95,21 +103,50 @@ const DEFAULT_MAINNET_CONNECTION_URL = 'https://api.mainnet-beta.solana.com';
...
-  isEnterprise?: boolean;
-}
 
 export class CandyShop extends BaseShop implements CandyShopAuctioneer, CandyShopEditionDropper {
   private _candyShopAddress: web3.PublicKey;
   private _isEnterprise: boolean;
   private _version: CandyShopVersion;
   private _program: Program | undefined;
 
+  static async initSolanaCandyShop(params: SolanaShopConstructorParams): Promise<CandyShop> {
```
Comment (haVincy) [inline core/sdk/src/shop/sol/CandyShop.ts:113]: `initSolCandyShop` ?

File: core/sdk/src/shop/sol/CandyShop.ts [hunk 3]
```diff
@@ -83,6 +88,9 @@ const DEFAULT_PRICE_DECIMALS_MIN = 0;
 const DEFAULT_VOLUME_DECIMALS = 1;
 const DEFAULT_VOLUME_DECIMALS_MIN = 0;
 const DEFAULT_MAINNET_CONNECTION_URL = 'https://api.mainnet-beta.solana.com';
+export interface SolanaShopConstructorParams extends BaseShopConstructorParams {
```
Comment (haVincy) [inline core/sdk/src/shop/sol/CandyShop.ts:91]: `SolShopConstructorParams` ?

File: core/sdk/src/shop/sol/CandyShop.ts (hunk 1 above)
Comment (RustySol) [reply core/sdk/src/shop/sol/CandyShop.ts:187]: yes will include this in the release note

**Pull request merged**